stdlib/printf.mli

```diff
@@ -175,6 +181,13 @@ val kbprintf : (Buffer.t -> 'd) -> Buffer.t ->
    @since 3.10.0
 *)
 
+val ikbprintf : (Buffer.t -> 'd) -> Buffer.t ->
+               ('a, Buffer.t, 'c, 'd) format4 -> 'a
```

Comment:
Same as above, we want to restrict `'c` to `unit` to keep the exact same type as `kbprintf`.